opts/envfile_test.go

```diff
@@ -0,0 +1,104 @@
+package opts
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// ParseEnvFile with a non existent file
+func TestParseEnvFileNonExistentFile(t *testing.T) {
+	_, err := ParseEnvFile("I don't exist as a file")
+	if err == nil {
+		t.Fatalf("Expected a PathError, got nothing")
+	}
+	if _, ok := err.(*os.PathError); !ok {
+		t.Fatalf("Expected a PathError, got [%v]", err)
+	}
+}
+
+// ParseEnvFile with an invalid file
+func TestParseEnvFileInvalidFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "tail-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	defer os.RemoveAll(f.Name())
```

Comment:
Minor thing but I think these defer's need to be switched so that we close it before we remove it. Defers are done in LIFO order.